src/sentry/api/base.py

```diff
@@ -202,6 +204,26 @@ def paginate(self, request, on_results=None, paginator_cls=Paginator,
 
         return Response(results, headers=headers)
 
+    def bail_on_xorg(self, request, **filters):
+        """
+        Will raise `ResourceDoesNotExist` when an authenticatee is attempting
+        to access a resource that is defined in another organization.
+        See: https://github.com/getsentry/sentry/issues/3807
+        """
+        if isinstance(filters, Organization):
```

Comment:
Sorry, I meant specifically, `filters` will never be anything other than a dict here. And if you attempted to pass a plain arg, it'd error.

You need to do something like:

```python
def foo(*args, **filters):
    if len(args):
        org = args[0]
````